web-client/src/presenter/actions/handleIdleLogoutAction.ts

```diff
@@ -1,35 +1,48 @@
+import { IDLE_LOGOUT_STATES } from '@shared/business/entities/EntityConstants';
 import { state } from '@web-client/presenter/app.cerebral';
 
 export const handleIdleLogoutAction = ({ get, path, store }: ActionProps) => {
   const constants = get(state.constants);
   const lastIdleAction = get(state.lastIdleAction);
   const idleLogoutState = get(state.idleLogoutState);
   const isUploading = get(state.fileUploadProgress.isUploading);
-  const user = get(state.user);
+  const userIsLoggedIn = !!get(state.token);
+  const clientNeedsToRefresh = get(state.clientNeedsToRefresh);
 
-  if (user && !isUploading && idleLogoutState.state === 'INITIAL') {
+  // Short-circuit here to prevent showing the "Are you still there?" modal, etc. when inappropriate
+  if (!userIsLoggedIn || clientNeedsToRefresh) {
```

Comment:
Originally, the "user is logged in?" check was short-circuiting, just like I have it here. Looking at the Git history, the check was accidentally moved into the first if block. After moving it back out, I also refactored the `if/else if` into multiple independent `if` blocks for easier readability.